internal/interceptor/outbound.go

```diff
@@ -64,3 +67,93 @@ type (
 	// multiple times on the same request.
 	StreamOutbound = middleware.StreamOutbound
 )
+
+var (
+	_ transport.UnaryOutbound  = UnaryOutboundFunc(nil)
+	_ transport.OnewayOutbound = OnewayOutboundFunc(nil)
+	_ transport.StreamOutbound = StreamOutboundFunc(nil)
+)
+
+// UnaryOutboundFunc defines a function type that implements the UnaryOutbound interface.
+type UnaryOutboundFunc func(context.Context, *transport.Request) (*transport.Response, error)
+
+// Call calls the UnaryOutbound function.
+func (f UnaryOutboundFunc) Call(ctx context.Context, req *transport.Request) (*transport.Response, error) {
+	return f(ctx, req)
+}
+
+// Start starts the UnaryOutbound function. This is a no-op in this implementation.
+func (f UnaryOutboundFunc) Start() error {
+	return nil
+}
+
+// Stop stops the UnaryOutbound function. This is a no-op in this implementation.
+func (f UnaryOutboundFunc) Stop() error {
+	return nil
+}
+
+// IsRunning returns whether the UnaryOutbound function is running. This is a no-op in this implementation.
+func (f UnaryOutboundFunc) IsRunning() bool {
+	return false
+}
+
+// Transports returns the transports used by the UnaryOutbound function. This is a no-op in this implementation.
+func (f UnaryOutboundFunc) Transports() []transport.Transport {
+	return nil
+}
+
+// OnewayOutboundFunc defines a function type that implements the OnewayOutbound interface.
+type OnewayOutboundFunc func(context.Context, *transport.Request) (transport.Ack, error)
```

Comment:
I don't think it's the best practice to convert function into the full interface, let's discuss it.